### ISSUE TITLE
[MNG-7783] Profile source point to file instead of to module

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#9204]
**[Slawomir Jaranowski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slawekjaranowski)** opened **[MNG-7783](https://issues.apache.org/jira/browse/MNG-7783?redirect=false)** and commented

Executing:

```
mvn help:active-profiles  -P run-its
```

With Maven 3.9.2 we have

```
The following profiles are active:


 - props (source: external)
 - run-its (source: external)
 - run-its (source: org.apache.maven.plugins:maven-clean-plugin:3.2.1-SNAPSHOT)
 - run-its (source: org.apache.maven.plugins:maven-plugins:38)
 - format-check (source: org.apache.maven:maven-parent:38) 
```

But with 4.x

```
The following profiles are active:


 - props (source: external)
 - run-its (source: external)
 - run-its (source: ..../maven-clean-plugin/pom.xml)
 - run-its (source: ..../.m2/repository-prv/org/apache/maven/plugins/maven-plugins/38/maven-plugins-38.pom)
 - format-check (source: ..../.m2/repository-prv/org/apache/maven/maven-parent/38/maven-parent-38.pom) 
```

---

**Affects:** 4.0.0-alpha-4, 4.0.0-alpha-5

**Issue Links:**
- [MRELEASE-1123](https://issues.apache.org/jira/browse/MRELEASE-1123) Maven Release plugin doesn't work in maven 4

**Remote Links:**
- [PR 1110
](https://github.com/apache/maven/pull/1110)

**Backported to:** [4.0.0-alpha-7](https://github.com/apache/maven/milestone/5?closed=1)

**Comment by jira-importer** [comment]
**[Guillaume Nodet](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gnt)** commented

This is caused by https://github.com/apache/maven/commit/9f88494b6064ad45ea2e2e1e3478afc7af0bc065

 

I've provided a PR to fix the problem, but the code is really fragile imho as using `toString()` to get an id with strong assomptions is a recipe for problems.

**Comment by jira-importer** [comment]
**[Guillaume Nodet](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gnt)** commented

Fixed with [         7fe9499       ](https://github.com/apache/maven/commit/7fe9499361c05f1c00dbda94d8f2a8df1c2c5f4e)
